### PR TITLE
[Refactor] move model.pretrained to model.backbone.init_cfg

### LINKED ISSUE #5318: [Refactor] Rewrite config files use init_cfg instead of model.pretrained
Using `model.pretrained` to Initialize the backbone will get a warning: *UserWarning: init_cfg without layer key if you do not define override key either, this init_cfg will do nothing*.
It is necessary to rewrite config files, using `init_cfg` to initialize the backbone.

### PR DESCRIPTION
**Pull request by BIGWangYuDong**
This PR move `model.pretrained` to `model.config.init_cfg` can solve the warning while loading the pretrained model. 

there are serval examples that we did in this PR:
1. We move `model.pretrained=XX` to `model.backbone.init_cfg` as:
```python
backbone=dict(
     arg1=xxx,
     arg2=xxx,
     init_cfg=dict(type='Pretrained', checkpoint='XXXX'))
```
2. When the config file need set `model.pretrained=None` such as in `config/cityscapes` , we rewrite the config file as:
```python
backbone=dict(
     arg1=xxx,
     arg2=xxx,
     init_cfg=None)
```

3. *[Need to check]*  In `config/selfup_pretrain`, while pretrained file is stored locally, we set `backbone.init_cfg=dict(type='Pretrained', checkpoint='./XX.pth')`.

Resolves 
#5318 
#5345

### CODE REVIEW

**Comment by hhaAndroid** [comment]
ci reported an error, it is indicating that some codes were not changed correctly.

**Review by hhaAndroid — COMMENTED** [review]
The detectors model has not been modified.

**Comment by BIGWangYuDong** [comment]
Have already merge the  master  branch and fix the errors in SSD-VGG backbone, I set the init_cfg priority > pretrained, using 
```python
if init_cfg is not None:
    self.init_cfg = init_cfg
else:
    if isinstance(pretrained, str):
        warnings.warn('DeprecationWarning: pretrained is deprecated, '
                              'please use "init_cfg" instead')
        self.init_cfg = [
                dict(type='Pretrained', checkpoint=pretrained)
          ]
    elif pretrained is None:
         self.init_cfg = [
              dict(type='Kaiming', layer='Conv2d'),
              dict(type='Constant', val=1, layer='BatchNorm2d'),
              dict(type='Normal', std=0.01, layer='Linear'),
         ]
    else:
           raise TypeError('pretrained must be a str or None')
```

File: mmdet/models/backbones/detectors_resnet.py [hunk 1]
```diff
@@ -238,9 +238,12 @@ def __init__(self,
                  pretrained=None,
                  init_cfg=None,
                  **kwargs):
-        assert init_cfg is None, 'To prevent abnormal initialization ' \
-                                 'behavior, init_cfg is not allowed to be set'
-        self.pretrained = pretrained
+        self.init_cfg = init_cfg
+        if pretrained is None:
```
Comment (BIGWangYuDong) [inline mmdet/models/backbones/detectors_resnet.py:242]: fix priority

File: mmdet/models/backbones/detectors_resnet.py [hunk 2]
```diff
@@ -238,9 +238,14 @@ def __init__(self,
                  pretrained=None,
                  init_cfg=None,
                  **kwargs):
-        assert init_cfg is None, 'To prevent abnormal initialization ' \
-                                 'behavior, init_cfg is not allowed to be set'
-        self.pretrained = pretrained
+        self.init_cfg = init_cfg
+        if init_cfg is not None:  # init_cfg priority > pretrained
+            self.pretrained = init_cfg['checkpoint']
```
Comment (hhaAndroid) [inline mmdet/models/backbones/detectors_resnet.py:243]: For insurance, it’s best not to take it directly

**Comment by BIGWangYuDong** [comment]
Add codes to ensure init_cfg can only initialize module by loading a pretrained model.
```python
assert isinstance(init_cfg, dict), 
     f'init_cfg must be a dict, but got {type(init_cfg)}'
if 'type' in init_cfg:
     assert init_cfg.get('type') == 'Pretrained', \
            'Only can initialize module by loading a pretrained model'
else:
      raise KeyError('`init_cfg` must contain the key "type"')
self.pretrained = init_cfg.get('checkpoint')
```

File: mmdet/models/backbones/ssd_vgg.py [hunk 3]
```diff
@@ -79,19 +79,26 @@ def __init__(self,
...
-        if isinstance(pretrained, str):
-            warnings.warn('DeprecationWarning: pretrained is deprecated, '
-                          'please use "init_cfg" instead')
-            self.init_cfg = [dict(type='Pretrained', checkpoint=pretrained)]
-        elif pretrained is None:
-            if init_cfg is None:
+
+        if init_cfg is not None:
+            self.init_cfg = init_cfg
+        else:
```
Comment (ZwwWayne) [inline mmdet/models/backbones/ssd_vgg.py:85]: should use elif here.

File: tests/test_models/test_backbones/test_detectors_resnet.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        frozen_stages=1,
+        norm_cfg=dict(type='BN', requires_grad=True),
+        norm_eval=True,
+        style='pytorch',
+        conv_cfg=dict(type='ConvAWS'),
+        sac=dict(type='SAC', use_deform=True),
+        stage_with_sac=(False, True, True, True),
+        output_img=True)
+    """Test init_weights config"""
+    with pytest.raises(AssertionError):
```
Comment (ZwwWayne) [inline tests/test_models/test_backbones/test_detectors_resnet.py:21]: Add comments indicating what error you are testing above each pytest.raises()

File: mmdet/models/backbones/detectors_resnet.py [hunk 5]
```diff
@@ -238,9 +238,22 @@ def __init__(self,
...
+        if init_cfg is not None:
+            assert isinstance(init_cfg, dict), \
+                f'init_cfg must be a dict, but got {type(init_cfg)}'
+            if 'type' in init_cfg:
+                assert init_cfg.get('type') == 'Pretrained', \
+                    'Only can initialize module by loading a pretrained model'
+            else:
+                raise KeyError('`init_cfg` must contain the key "type"')
+            self.pretrained = init_cfg.get('checkpoint')
+        else:
```
Comment (hhaAndroid) [inline mmdet/models/backbones/detectors_resnet.py:252]: Can simplify logic

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**